src/nafill.c

```diff
@@ -113,7 +113,7 @@ SEXP nafillR(SEXP obj, SEXP type, SEXP fill, SEXP nan_is_na_arg, SEXP inplace, S
   if (obj_scalar) {
     if (binplace)
       error(_("'x' argument is atomic vector, in-place update is supported only for list/data.table"));
-    else if (!isReal(obj) && !isInteger(obj))
+    else if (!isReal(obj) && !isInteger(obj) && !isLogical(obj))
       error(_("'x' argument must be numeric type, or list/data.table of numeric types"));
```

Comment:
should we mention logical here?

Reply:
True, I'd missed all documentation. PTAL